solver/simple.go

```diff
@@ -426,62 +444,30 @@ func (f *parallelGuard) acquire(ctx context.Context, d string) (<-chan struct{},
 	return ch, closer
 }
 
-type resultCache interface {
-	set(ctx context.Context, key string, r Result) error
-	get(ctx context.Context, key string) (Result, bool, error)
-}
-
-type inMemCache struct {
-	cache map[string]Result
-	mu    sync.Mutex
-}
-
-func newInMemCache() *inMemCache {
-	return &inMemCache{cache: map[string]Result{}}
+// RefIDStore uses a BoltDB database to store links from computed cache keys to
+// worker ref IDs.
+type RefIDStore struct {
```

Comment:
I split the ref ID storage out from the result loader so that the former can be used in the solver &  `worker/simple.go`.